spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java

```diff
@@ -0,0 +1,283 @@
+		super.onInit();
+		Assert.notNull(this.headerMapper, "Header mapper can not be null!");
+
+		if (this.threadFactory == null) {
+			this.threadFactory = new CustomizableThreadFactory(getComponentName() + "-thread-");
+		}
+
+		this.executorService = Executors.newSingleThreadExecutor(this.threadFactory);
+
+		if (!this.enableBatch) {
+			this.debeziumEngineBuilder.notifying(new StreamChangeEventConsumer<byte[]>());
+		}
+		else {
+			this.debeziumEngineBuilder.notifying(new BatchChangeEventConsumer<byte[]>());
+		}
+
+		this.debeziumEngine = this.debeziumEngineBuilder.build();
+	}
+
+	@Override
+	protected void doStart() {
+		if (this.latch.getCount() > 0) {
+			return;
+		}
+		this.latch = new CountDownLatch(1);
+		this.executorService.execute(() -> {
+			try {
+				// Runs the debezium connector and deliver database changes to the registered consumer. This method
+				// blocks until the connector is stopped.
+				// If this instance is already running, then the run immediately returns.
+				// When run the connector and starts polling the configured connector for change events.
+				// All messages are delivered in batches to the consumer registered with this debezium engine.
+				// The batch size, polling frequency, and other parameters are controlled via connector's configuration
+				// settings. This continues until this connector is stopped.
+				// This method can be called repeatedly as needed.
+				this.debeziumEngine.run();
+			}
+			finally {
+				this.latch.countDown();
+			}
+		});
+	}
+
+	@Override
+	protected void doStop() {
+		try {
+			this.debeziumEngine.close();
+		}
+		catch (IOException e) {
+			logger.warn(e, "Debezium failed to close!");
+		}
+		try {
+			if (!this.latch.await(5, TimeUnit.SECONDS)) {
+				throw new IllegalStateException("Failed to stop " + this);
+			}
+		}
+		catch (InterruptedException ignored) {
+		}
+	}
+
+	@Override
+	public void destroy() {
+		super.destroy();
+
+		this.executorService.shutdown();
+		try {
+			this.executorService.awaitTermination(5, TimeUnit.SECONDS);
+		}
+		catch (InterruptedException e) {
+			throw new IllegalStateException("Debezium failed to close!", e);
+		}
+	}
+
+	private <T> Message<?> toMessage(ChangeEvent<T, T> changeEvent) {
+
+		Object key = changeEvent.key();
+		Object payload = changeEvent.value();
+		String destination = changeEvent.destination();
+
+		// When the tombstone event is enabled, Debezium serializes the payload to null (e.g. empty payload)
+		// while the metadata information is carried through the headers (debezium_key).
+		// Note: Event for none flattened responses, when the debezium.properties.tombstones.on.delete=true
+		// (default), tombstones are generate by Debezium and handled by the code below.
+		if (payload == null && DebeziumMessageProducer.this.enableEmptyPayload) {
+			payload = Optional.empty();
+		}
+
+		// If payload is still null ignore the message.
+		if (payload == null) {
+			logger.info("Dropped null payload message");
```

Comment:
Can we also add here something like `" for:" + changeEvent`?
To have some traceability in the logs to correlate such a not with an event produced by the Debezium.